osrf_testing_tools_cpp/cmake/osrf_testing_tools_cpp_extract_and_build_googletest.cmake

```diff
@@ -72,9 +72,7 @@ macro(osrf_testing_tools_cpp_extract_and_build_googletest
     # (see https://wiki.gentoo.org/wiki/AddressSanitizer/Problems#pthread_linking_issues
     # for some additional information).  To work around that, we unconditionally
     # add the -pthread flag for Linux machines so it will always work
-    execute_process(COMMAND bash "-c" "sed -i 's@\${CMAKE_THREAD_LIBS_INIT}@\${CMAKE_THREAD_LIBS_INIT} -pthread@' ${__prefix}-src/googletest/cmake/internal_utils.cmake"
-      RESULT_VARIABLE result
-      WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/${__prefix}-extracted)
+    link_libraries("-pthread")
```

Comment:
Hmm, seems like this change makes it no longer agree with the comments above?

@clalancette you added this logic to fix asan building in the past, can you comment?

Reply:
> Hmm, seems like this change makes it no longer agree with the comments above?
> 
> @clalancette you added this logic to fix asan building in the past, can you comment?

It still agrees with the comment.  That is, asan still has the problem that it exposes a "fake" pthread that is good enough to fool CMake into thinking it found it, but it isn't good enough to successfully link.

What this change does is take advantage of the fact that this is a single CMake context, so using `link_libraries("-pthread")` works.  In my testing, this still links properly when compiling with asan, so :+1: from me.

Reply:
Ok, good enough for me. Thanks.